### PR TITLE
executor: fix bug when auto_increment meets unsigned.

### LINKED ISSUE #4817: select return empty when using non-primary key auto_increment column as where condition
1. What did you do?

```
mysql> CREATE TABLE test(id int(10) UNSIGNED NOT NULL AUTO_INCREMENT, p int(10) UNSIGNED NOT NULL, PRIMARY KEY(p), KEY(id));     
Query OK, 0 rows affected (0.16 sec)

mysql> show create table test;
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                         |
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| test  | CREATE TABLE `test` (
  `id` int(10) UNSIGNED NOT NULL AUTO_INCREMENT,
  `p` int(10) UNSIGNED NOT NULL,
  PRIMARY KEY (`p`),
  KEY `id` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> insert into test(p) value(1);
Query OK, 1 row affected (0.03 sec)

mysql> select * from test;
+----+---+
| id | p |
+----+---+
|  1 | 1 |
+----+---+
1 row in set (0.00 sec)

mysql> select * from test where p=1;
+----+---+
| id | p |
+----+---+
|  1 | 1 |
+----+---+
1 row in set (0.00 sec)

mysql> select * from test where id=1;
Empty set (0.00 sec)
```

2. What did you expect to see?
```
mysql> select * from test where id=1;
+----+---+
| id | p |
+----+---+
|  1 | 1 |
+----+---+
1 row in set (0.00 sec)
```

3. What did you see instead?
```
mysql> select * from test where id=1;
Empty set (0.00 sec)
```

4. What version of TiDB are you using (`tidb-server -V`)?
```
Release Version: 1.0.0
Git Commit Hash: f4e08c15f41f52475184a181ea597ee1cc855c90
Git Commit Branch: release-1.0
UTC Build Time:  2017-10-16 05:32:55
```

Use id as primary key instead then is fine:
```
mysql> CREATE TABLE test(id int(10) UNSIGNED NOT NULL AUTO_INCREMENT, p int(10) UNSIGNED NOT NULL, PRIMARY KEY(id), KEY(p));
Query OK, 0 rows affected (0.16 sec)

mysql> show create table test;
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                        |
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| test  | CREATE TABLE `test` (
  `id` int(10) UNSIGNED NOT NULL AUTO_INCREMENT,
  `p` int(10) UNSIGNED NOT NULL,
  PRIMARY KEY (`id`),
  KEY `p` (`p`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> insert into test(p) value(1);
Query OK, 1 row affected (0.02 sec)

mysql> select * from test;
+----+---+
| id | p |
+----+---+
|  1 | 1 |
+----+---+
1 row in set (0.00 sec)

mysql> select * from test where id=1;
+----+---+
| id | p |
+----+---+
|  1 | 1 |
+----+---+
1 row in set (0.01 sec)

mysql> select * from test where p=1;
+----+---+
| id | p |
+----+---+
|  1 | 1 |
+----+---+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by winoros**
fix #4817 
This bug will make the index key encoded to wrong value and make range query failed.
PTAL @shenli @coocood @hanfei1991

### CODE REVIEW

File: executor/write.go [hunk 1]
```diff
@@ -1107,6 +1111,11 @@ func (e *InsertValues) adjustAutoIncrementDatum(row []types.Datum, i int, c *tab
 			return errors.Trace(err)
 		}
 		e.ctx.GetSessionVars().InsertID = uint64(recordID)
+		if mysql.HasUnsignedFlag(c.Flag) {
+			row[i].SetUint64(uint64(recordID))
+		} else {
+			row[i].SetInt64(recordID)
+		}
 		row[i].SetInt64(recordID)
```
Comment (hanfei1991) [inline executor/write.go]: remove this

Comment (coocood) [inline executor/write.go]: This line should be removed?

Comment (coocood) [reply executor/write.go]: Should be covered by test.

Comment (winoros) [reply executor/write.go]: covered.

**Pull request merged**